components/style/properties/longhand/text.mako.rs

```diff
@@ -175,34 +177,40 @@ ${helpers.single_keyword("unicode-bidi",
         SpecifiedValue::empty()
     }
     /// none | [ underline || overline || line-through || blink ]
-    pub fn parse(_context: &ParserContext, input: &mut Parser) -> Result<SpecifiedValue, ()> {
+    pub fn parse<'i, 't>(_context: &ParserContext, input: &mut Parser<'i, 't>)
+                         -> Result<SpecifiedValue, ParseError<'i>> {
         let mut result = SpecifiedValue::empty();
         if input.try(|input| input.expect_ident_matching("none")).is_ok() {
             return Ok(result)
         }
         let mut empty = true;
 
-        while input.try(|input| {
-                if let Ok(ident) = input.expect_ident() {
-                    match_ignore_ascii_case! { &ident,
-                        "underline" => if result.contains(UNDERLINE) { return Err(()) }
-                                       else { empty = false; result.insert(UNDERLINE) },
-                        "overline" => if result.contains(OVERLINE) { return Err(()) }
-                                      else { empty = false; result.insert(OVERLINE) },
-                        "line-through" => if result.contains(LINE_THROUGH) { return Err(()) }
-                                          else { empty = false; result.insert(LINE_THROUGH) },
-                        "blink" => if result.contains(BLINK) { return Err(()) }
-                                   else { empty = false; result.insert(BLINK) },
-                        _ => return Err(())
+        loop {
+            let result: Result<_, ParseError> = input.try(|input| {
+                match input.expect_ident() {
+                    Ok(ident) => {
+                        (match_ignore_ascii_case! { &ident,
```

Comment:
Fixing this will be kind of annoying given how often this pattern appears. I would prefer not to, if that's alright.